cmd/sink/main.go

```diff
@@ -25,24 +25,28 @@ import (
 )
 
 const (
-	DeleteWhenEnvVar = "DELETE_WHEN"
+	mountPathEnvVar = "MOUNT_PATH"
 )
 
 type Sink struct {
 	Db          database.DBInterface
-	DeleteWhen  string
 	EventClient ceClient.Client
+	Filters     *expr.Filters
 	K8sClient   *dynamic.DynamicClient
 	Logger      *log.Logger
 }
 
-func NewSink(db database.DBInterface, logger *log.Logger) *Sink {
+func NewSink(db database.DBInterface, logger *log.Logger, filters *expr.Filters) *Sink {
 	if logger == nil {
 		logger = log.New(os.Stderr, "", log.LstdFlags|log.Lmicroseconds|log.LUTC)
-		logger.Println("sink was provided a nil logger, falling back to defualt logger")
+		logger.Println("Sink was provided a nil logger, falling back to default logger")
 	}
 	if db == nil {
-		logger.Fatalln("cannot start sink when db connection is nil")
+		logger.Fatalln("Cannot start sink when db connection is nil")
+	}
+	if filters == nil {
+		logger.Println("Sink was provided nil filters. Archive and delete operations will not happen until new filters are provided")
+		filters = expr.EmptyFilters()
```

Comment:
This won't be true because the expr.NewFilters always return a filters object, even if there are errors.

Reply:
I agree, but this is more of a defensive thing. Since the method takes in a `*Filter` it is possible a bug is introduced that causes filters to be `nil` here. Having the sink log this instead of crashing will make discovering the bug easier

Reply:
If its important enough to create a check here, I think its important enough to have it tested. If we test that the method does not return a `nil` no matter what, we don't need this check. If this does not convince you, then keep it as is.